src/todo.js

```diff
@@ -0,0 +1,98 @@
+export default class Todo {
+  static count = 0;
+
+  constructor(completed, description) {
+    // eslint-disable-next-line
+    this.index = this.constructor.count++;
+    this.completed = completed;
+    this.description = description;
+  }
+
+  getAllTodos = () => {
+    const todos = JSON.parse(localStorage.getItem('todos'));
+    return todos || [];
+  }
+
+  addTodo = (todo) => {
+    const todos = this.getAllTodos();
+    todos.unshift(todo);
+
+    localStorage.setItem('todos', JSON.stringify(todos));
+
+    return todos;
+  }
+
+  renderTodo = () => {
+    const todos = this.getAllTodos();
+
+    const todosListEl = document.getElementById('todos-list');
+    todosListEl.innerHTML = '';
+
+    todos.forEach((todo) => {
+      todosListEl.innerHTML += `
+        <div class="todo" id=${todo.index}>
+        
+          <div class="todoTask">
+            <input type="checkbox" id="todoCheck" name="todoCheck"> 
+            <input type="text" id="description" class="checkboxes" for="todoCheck" data-editID="${todo.index}" value="${todo.description}">
+           
+
+          </div>
+
+          <div class="options">
+
+             <button id ="dropdown" class="btnTogle"><i class="drag bi bi-three-dots-vertical"></i></button>
+
+            <div id="sectiontohide" class="optionDropdown">
+                <button id="delete" class="option-item"><i class="bi bi-trash3" data-index="${todo.index}"></i></button>
+            </div>
+
+         </div>
+        </div>
+       
+        `;
+
+      if (todo.completed) {
+        document.getElementById('todoCheck').checked = true;
+      }
+    });
+
+    const deleteBtn = document.querySelectorAll('#delete');
+    const textLabel = document.querySelectorAll('#description');
+    const dropdownBtn = document.querySelectorAll('#dropdown');
+    const divDropdownBtn = document.querySelectorAll('#sectiontohide');
+
+    deleteBtn.forEach((button) => {
+      button.addEventListener('click', (event) => {
+        const id = parseInt(event.target.getAttribute('data-index'), 10);
+        let localStoragetodos = this.getAllTodos();
+        localStoragetodos = localStoragetodos.filter((todo) => todo.index !== id);
+        // eslint-disable-next-line
+        localStorage.setItem('todos', JSON.stringify(localStoragetodos));
+        // eslint-disable-next-line
+        location.reload();
+      });
+    });
```

Comment:
- [ ] In general. your application functions well 👏. However, I realized that **when you delete a task, the indexes of the remaining tasks are not updated**. Consequently, **if you try to add another task, it will have an index of on**e. Please implement the functionality to update tasks' indexes when one of them is deleted as it is in the project requirements. 😃